internal/hcops/server.go

```diff
@@ -117,22 +147,18 @@ func (c *AllServersCache) getCache(getSrv func() (*hcloud.Server, bool)) (*hclou
 				continue
 			}
 			if _, ok := c.byPrivIP[n.IP.String()]; ok {
-				klog.Warningf("Overriding AllServersCache entry for private ip %s with server %s\n", n.IP.String(), srv.Name)
+				klog.Warningf("Overriding AllServersCache entry for private ip %s with server %s\n", n.IP.String(), server.Name)
 			}
-			c.byPrivIP[n.IP.String()] = srv
+			c.byPrivIP[n.IP.String()] = server
 		}
 
 		// Index servers by their names.
-		c.byName[srv.Name] = srv
+		c.byName[server.Name] = server
```

Comment:
```suggestion
		c.byName[server.Name] = servers[i]
```

Is there a chance that are storing the loop reference every time?  Shouldn't we prefer the indexed loop to make it explicit?

Reply:
I do not think we were affected by this, otherwise the whole thing would not work at all. But we do not have to care about this anymore, the language was changed in Go 1.22 to avoid this issue altogether: https://go.dev/doc/go1.22#language